server/src/test/java/org/elasticsearch/ingest/PipelineFactoryTests.java

```diff
@@ -53,6 +53,32 @@ public void testCreate() throws Exception {
         assertThat(pipeline.getProcessors().get(1).getTag(), nullValue());
     }
 
+    public void testContainsNestedProcessors() throws Exception {
+        // create a pipeline with non-nested processors
+        Map<String, Object> processorConfig0 = new HashMap<>();
```

Comment:
I'd drop `processorConfig0` and `processorConfig1`, I don't think they carry their weight as variables. Probably just replace `new HashMap<>()` at each of their current uses.